staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go

```diff
@@ -1137,9 +1137,11 @@ func (c *Cacher) dispatchEvent(event *watchCacheEvent) {
 		// it unconditionally to ensure safety and reduce deep-copying.
 		//
 		// Make a shallow copy to allow overwriting Object and PrevObject.
-		wcEvent := *event
-		setCachingObjects(&wcEvent, c.versioner)
-		event = &wcEvent
+		if _, ok := event.Object.(*cachingObject); !ok {
```

Comment:
It's not clear to me what code path leads to an event being sent to disatchEvent which *already* contains a caching object.

Reply:
@liggitt Sorry, I forgot to submit a portion of the code. I changed the watchCache sliding window event to Cachingobject